kube-core/src/validation.rs

```diff
@@ -0,0 +1,118 @@
+//! CEL validation for CRDs
+
+use std::str::FromStr;
+
+#[cfg(feature = "schema")] use schemars::schema::Schema;
+use serde::{Deserialize, Serialize};
+
+/// Rule is a CEL validation rule for the CRD field
+#[derive(Default, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct Rule {
```

Comment:
note to self; this may perhaps be churned through a cel crate against an object to perform validation client side. should investigate this later.